src/controllers/hid/hidcontroller.cpp

```diff
@@ -208,8 +220,22 @@ bool HidController::poll() {
             return false;
         } else if (bytesRead == 0) {
             // No packet was available to be read
+            for (int rptIdx = 0; rptIdx <= kMaxHidInputReportId; rptIdx++) {
```

Comment:
This cycles 255 loop will run on each poll, even if we have not received a report. 
Can we check a simple counter in addition as a short cut?

Reply:
I changed the algorithm to simple counters, which no longer require an expensive loop.